lang/cs/Org.Apache.REEF.IMRU/API/IMRUJobDefinitionBuilder.cs

```diff
@@ -203,6 +211,17 @@ public IMRUJobDefinitionBuilder SetUpdateTaskMemory(int memory)
         }
 
         /// <summary>
+        /// Sets Per Map Configuration
```

Comment:
> you want to have ISet<IPerMapperConfig> ??

Yes, as a constructor parameter /  attribute of `IMRUDriver`

> what use will it have?

  * Different creators of `IPerMapConfiguration` implementations do not have to coordinate. It is up to the user to add the configuration of those to the Driver Configuration.
  * We get early checking of the various configurations for the `IPerMapConfiguration` implementations.
  * `IMRUDriver` becomes much simpler, as Tang will inject an empty `ISet` if no `IPerMapConfiguration` is bound.

> user can always merge and set it as single configuration or am I missing something?

User's can't merge the configurations, as they are only created on the Driver side. And so far, we only allow users of IMRU to assemble the IMRU Job, not to take part in its driver. Hence, the situation where two libraries want to create additional per-Task configurations cannot be implemented with a single `IPerMapConfiguration` instance. However, it seems likely that we'll have that situation soon enough: One for the block manager, one for the buffer manager, one for specifics of group com, ...